scipy/optimize/tests/test_minimize_constrained.py

```diff
@@ -676,3 +676,64 @@ def opt(x):
         A = np.matrix(np.diag([1, 1]))
     lin_cons = LinearConstraint(A, -1, np.inf)
     minimize(opt, 2*[1], constraints = lin_cons)  # just checking that there are no errors
+
+
+class TestBoundedNelderMead:
+
+    @pytest.mark.parametrize('bounds',
+                             [Bounds(-np.inf, np.inf),
+                              Bounds(-np.inf, -0.8),
+                              Bounds(3.0, np.inf),
+                              Bounds([3.0, 1.0], [4.0, 5.0]),
+                              ])
+    def test_rosen_brock_with_bounds(self, bounds):
+        prob = Rosenbrock()
+        with suppress_warnings() as sup:
+            sup.filter(UserWarning, "Initial guess is not within "
+                                    "the specified bounds")
+            result = minimize(prob.fun, [-10, -10],
+                              method='Nelder-Mead',
+                              bounds=bounds)
+            assert np.less_equal(bounds.lb, result.x).all()
+            assert np.less_equal(result.x, bounds.ub).all()
+            assert np.allclose(prob.fun(result.x), result.fun)
```

Comment:
Why is this relevant here?

Reply:
>a test (with bounds) to show a normal usage with normal convergence to the known solution.

I added known solutions and assert to check the result is close or not. PTAL,

>Why is this relevant here?

The current Nelder-mead method is handling each simprex and its objective value using different two variables.
https://github.com/scipy/scipy/blob/08390dcde02d048a4a1b07a132510a9d0ea6f28c/scipy/optimize/optimize.py#L795-L796
So, if we don't update both variables correctly when clipping, `result.fun` might be wrong.
This test check it: `assert np.allclose(prob.fun(result.x), result.fun)`